packages/shared/src/components/auth/AuthOptions.tsx

```diff
@@ -239,70 +238,67 @@ function AuthOptions({
     onSetActiveDisplay(AuthDisplay.CodeVerification);
   };
 
-  useWindowEvents<SocialRegistrationFlow>(
-    'message',
-    AuthEvent.SocialRegistration,
-    async (e) => {
-      if (e.data?.flow) {
-        const connected = await getKratosFlow(
-          AuthFlow.Registration,
-          e.data.flow,
-        );
+  useEventListener(globalThis, 'message', async (e) => {
```

Comment:
This one will require some manual/preview verification before we launch it.
Mostly on both email registration and social, just to exclude edge-cases.

Reply:
Yes!

Reply:
Will this be attached to the window with just this?

Reply:
Yes, and using `globalThis` nomenclature instead of window as we do recently.

Reply:
I tested the cases above (updated description).